### PR TITLE
Ml add more aggregate stats

### PR DESCRIPTION
**Pull request by magneland**
This adds sub-totals for code and tests stats to the output.

### CODE REVIEW

**Comment by bleonard** [comment]
I missed this (for 3 years!)
Want to still merge this in?

@etagwerker seems to be using this. Want to test it out?

**Review by etagwerker — COMMENTED** [review]
@magneland Would you be open to adding more tests for this new behavior? 

I'm having a hard time understanding what it does (I'm new to this codebase and I'd like to help)

File: lib/rails_stats/code_statistics.rb [hunk 1]
```diff
@@ -15,18 +15,18 @@ def initialize(root_directory)
...
       @statistics.each { |key, stats| print_line(key, stats) }
       print_splitter
 
-      if @total
-        print_line("Total", @total)
-        print_splitter
-      end
+      print_line("Code", @code_total)
+      print_line("Tests", @tests_total)
+      print_line("Total", @grand_total)
```
Comment (magneland) [inline lib/rails_stats/code_statistics.rb:28]: This is the main change.
The output changes from:
```
| Total                | 64560 | 51668 |     685 |    4354 |   6 |     9 |
```
to:
```
| Code                 |   17640 |   13748 |     338 |    1557 |   4 |     6 |
| Tests                |   46920 |   37920 |     347 |    2797 |   8 |    11 |
| Total                |   64560 |   51668 |     685 |    4354 |   6 |     9 |
```

**Comment by magneland** [comment]
> @magneland Would you be open to adding more tests for this new behavior?
> 
> I'm having a hard time understanding what it does (I'm new to this codebase and I'd like to help)

For reference here is the original `railties` implementation:
https://github.com/rails/rails/blob/55badbba3be76172a10519488c2a2747a0552728/railties/lib/rails/code_statistics.rb

So this codebase actually has zero tests, and I think that is actually OK because it's a command line tool that prints a bit of information about a Rails app.

If this is still useful I think it should be sufficient to just run this branch and master branch on various repos and diff the output.

**Comment by etagwerker** [comment]
@magneland Cool, I just tested this in a sample application and it looks good. 👍 

@bleonard I think this would be a good addition to the project.

**Review by etagwerker — APPROVED** [review]

**Comment by etagwerker** [comment]
@magneland Thanks for your contribution! @bleonard just gave me access to merge this. 😄

**Pull request merged**